cmd/kube-rbac-proxy/app/kube-rbac-proxy.go

```diff
@@ -294,36 +283,52 @@ func secureServerRunner(
 		<-listenerStoppedCh
 		<-stoppedCh
 		return err
-	}
+	})
 
-	interrupter := func(err error) {
+	interrupter := interrupterFunc(func(err error) {
 		serverCtxCancel()
-	}
+	})
 
 	return runner, interrupter
 }
 
-func setupAuthorizer(krbInfo *server.KubeRBACProxyInfo, delegatedAuthz *serverconfig.AuthorizationInfo) (authorizer.Authorizer, error) {
-	staticAuthorizer, err := static.NewStaticAuthorizer(krbInfo.Authorization.Static)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create static authorizer: %w", err)
+func setupAuthorizer(krpInfo *server.KubeRBACProxyInfo, delegatedAuthz *serverconfig.AuthorizationInfo) (authorizer.Authorizer, error) {
+	authz := delegatedAuthz.Authorizer
+
+	if len(krpInfo.Authorization.Static) > 0 {
+		staticAuthorizer, err := static.NewStaticAuthorizer(krpInfo.Authorization.Static)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create static authorizer: %w", err)
+		}
+		authz = union.New(staticAuthorizer, authz)
 	}
 
-	var authz authorizer.Authorizer = rewrite.NewRewritingAuthorizer(
-		union.New(
-			staticAuthorizer,
-			delegatedAuthz.Authorizer,
-		),
-		krbInfo.Authorization.RewriteAttributesConfig,
-	)
+	if krpInfo.Authorization.RewriteAttributesConfig != nil {
```

Comment:
I mentioned this to you in Slack, but this introduces a regression. Before doing that change, we would need to do the following:
- What happens to requests, if this piece of code is not run every time?
  - Most probably we won't mask the authorizer.AttirbutesRecord to be a non-resource-request (and some more params).
- IF we don't  do that, what would happen to our requests? Would they default to resource requests? This would most probably not work.

Reply:
As we discussed, I dropped commit a24945c11cd8fef7f01e56924a91281b6211fe1e from this PR as we need to tackle this separately and differently.